allen_exporter/exporter.py

```diff
@@ -0,0 +1,416 @@
+                      dff_table, event_table, eye_tracking_table, path):
+
+    # Screen: mean and std of pixel values, weighted by image distribution
+    stimulus_images = stimulus_table[stimulus_table['image_name'].str.contains('im', case=False, na=False)]
+    images_mean = stimulus_templates['warped'].sort_index().apply(np.mean)
+    images_std = stimulus_templates['warped'].sort_index().apply(np.std)
+
+    counts = stimulus_images['image_name'].value_counts().sort_index()
+    nr_rows = counts.sum()
+
+    weights = counts / nr_rows
+    mean_pixel = np.sum(images_mean * weights)
+    std_pixel = np.sqrt(np.sum((images_std**2 + (images_mean - mean_pixel)**2) * weights))
+
+    np.save(os.path.join(path, 'screen/meta/means.npy'), mean_pixel)
+    np.save(os.path.join(path, 'screen/meta/stds.npy'), std_pixel)
+
+    # Treadmill: average and std of running speed
+    mean_speed = running_speed_table['speed'].mean()
+    std_speed = running_speed_table['speed'].std()
+
+    np.save(os.path.join(path, 'treadmill/meta/means.npy'), mean_speed)
+    np.save(os.path.join(path, 'treadmill/meta/stds.npy'), std_speed)
+
+    # dF/F and events: average and std
+    neurons = np.stack(dff_table['dff'].values)
+
+    # Compute mean and std along axis 0 (i.e., for each position 0–11)
+    mean_values = np.mean(neurons, axis=1)
+    std_values = np.std(neurons, axis=1)
+
+    np.save(os.path.join(path, 'responses/meta/means.npy'), np.array([mean_values]))
+    np.save(os.path.join(path, 'responses/meta/stds.npy'), np.array([std_values]))
+
+    # Eye tracker: mean and std of all columns (except timestamps)
+    eye_tracking_data = eye_tracking_table.drop(columns='timestamps', errors='ignore')
+    mean_values = eye_tracking_data.mean()
+    std_values = eye_tracking_data.std()
+
+    np.save(os.path.join(path, 'eye_tracker/meta/means.npy'), mean_values.to_numpy())
+    np.save(os.path.join(path, 'eye_tracker/meta/stds.npy'), std_values.to_numpy())
+
+    
+
+# function to save all the images once
+def save_images(stimulus_table, stimulus_templates, output_dir, compressed=True):
+    images = stimulus_templates['warped']
+    idxs = images.index
+        
+    for i, image in enumerate(images):
+        idx = idxs[i]
+        npy_path = os.path.join(output_dir, f"{idx}.npy")
+        np.save(npy_path, image)
+
+    # save all movies used
+    mv_names = stimulus_table[stimulus_table['stimulus_block_name'].str.contains('movie', case=False, na=False)]['stimulus_block_name'].unique()
+
+    if compressed:
+        for name in mv_names:
+            movie = np.load(f'../data/movies/{name}.npy')
+            output_path = os.path.join(output_dir, f"{name}")
+            grayscale_to_rgb_video(movie, output_path, fps=30, crf=23)
+            
+    else:
+        for name in mv_names:
+            movie = np.load(f'../data/movies/{name}.npy')
+            npy_path = os.path.join(output_dir, f"{name}.npy")
+            np.save(npy_path, movie)
```

Comment:
To avoid code duplicate, make it more concise and readable 
```suggestion
    for name in mv_names:
        movie = np.load(f’../data/movies/{name}.npy’)
        output_path = os.path.join(output_dir, f”{name}“)
        if compressed:
            grayscale_to_rgb_video(movie, output_path, fps=30, crf=23)
        else:
            np.save(f”{output_path}.npy”, movie)
```